### PR TITLE
Plane and AABB added

### PR DESCRIPTION
**Pull request by CedNaru**
Plane CoreType added
AABB CoreType added
godot.def updated with matching memory layout.

### CODE REVIEW

**Review by piiertho — APPROVED** [review]
Great work ! Better than #146, which I close.

File: godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+        }
+    }
+
+    internal constructor(mem: COpaquePointer) : this() {
+        this.setRawMemory(mem)
+    }
+
+
+    //INTEROP
+    //INTEROP
```
Comment (raniejade) [inline godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt]: nit: duplicate line?

File: godot-kotlin/godot-library/src/nativeInterop/cinterop/godot.def [hunk 2]
```diff
@@ -27,4 +27,14 @@ typedef  struct godot_quat_layout {
     real_t y;
     real_t z;
     real_t w;
-} godot_quat_layout;
\ No newline at end of file
+} godot_quat_layout;
+
+typedef  struct godot_plane_layout {
```
Comment (raniejade) [inline godot-kotlin/godot-library/src/nativeInterop/cinterop/godot.def]: nit: extra space

File: godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+
+    //UTILITIES
+    override fun equals(other: Any?): Boolean =
+        when (other) {
+            is AABB -> (position == other.position && size == other.size)
+            else -> false
+        }
+
+    override fun toString(): String =
+        "$position - $size"
```
Comment (raniejade) [inline godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt]: Can we make this `AABB(position=<position>, size=<size>)`? I find it more easier when debugging when the class name and its properties is included.

File: godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Plane.kt [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+     * Returns the orthogonal projection of point p into a point in the plane.
+     */
+    fun project(point: Vector3): Vector3 {
+        return point - normal * distanceTo(point)
+    }
+
+
+    //UTILITIES
+    override fun toString(): String {
+        return "${this.normal.toString()},$d"
```
Comment (raniejade) [inline godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Plane.kt]: same as the comment above, `Plane(normal=<normal>, d=<d>)`

File: godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt (hunk 3 above)
Comment (CedNaru) [reply godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt]: https://github.com/godotengine/godot/blob/c304b66a68f539850723126db7cd0286aba99fa7/core/math/aabb.cpp#L399

I'm using the toString format of the Cpp object.  We can do our own thing if you wish.

File: godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Plane.kt (hunk 4 above)
Comment (CedNaru) [reply godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Plane.kt]: same as previous comment

File: godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt (hunk 3 above)
Comment (chippmann) [reply godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/AABB.kt]: I agree with @raniejade here.

File: godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Plane.kt (hunk 4 above)
Comment (chippmann) [reply godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Plane.kt]: Also agree with @raniejade

**Review by chippmann — APPROVED** [review]

**Pull request merged**